power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/measured_values.hpp

```diff
@@ -15,6 +15,15 @@ Collect all measured Values
 
 namespace power_grid_model::math_solver {
 
+namespace detail {
+template <bool sym> inline RealValue<sym> cabs_or_real(ComplexValue<sym> const& value) {
+    if (is_nan(imag(value))) {
+        return real(value); // only keep real part
+    }
+    return cabs(value); // get abs of the value
```

Comment:
this logic should go OK because there's no division involved. `imag == 0` happens so rarely that even just evaluating the `if` statement every call would be overhead (branch prediction). since there's no division before taking the square of the norm, this will always be positive (by IEEE-754 floating point standard) regardless of architecture